### DISCUSSION TITLE
Need help re on/off state coloring

### DISCUSSION THREAD

**Discussion by willliamchan** [Clooos/Bubble-Card#512] [Questions about config, custom styles and templates] [answered]
![image](https://github.com/Clooos/Bubble-Card/assets/54702221/9f2cf0c6-c610-45af-bca5-4dc107997ed4)

So I have two aircons using bubble card and I'd like the color stays just like the screenshot when it's "Off" and when it is "On" (It could be heating or cooling) and I'd like it to pick the default "on" color from the theme just like the "Fan" entity. Is it something that I can do using style please?

Thank you!

**Comment by Clooos** [comment]
Hi! Is your fan a light entity? Because this color is only for light entities, your fan card should be blue on my theme.

**Reply by willliamchan** [reply]
light entity - no, it's a fan entity and i add it as a switch in bubble card, I am using your theme yaml but I had the colors updated.

In fact I tried not to raise up too many request and if i can have the style code example i can play with it, as besides aircon i got quite a few radiators that i wanna do the same.

Also when it comes to locks and contact sensors, i'd prefer the color is dimmed when it is closed and light up when it's open/unlocked.

While you give yourself in so quickly, I in fact want to ask if you can make another theme that is more for older people with bad eyesight too

**Reply by Clooos** [reply]
If you changed it that makes sense, for the color of the button/icon you can change it with a custom style.

Here is how :

```yaml
.bubble-button-card-container {
  background: ${hass.states['fan.your_entity'].state === 'state 1' ? 'color if state 1' : 
  hass.states['fan.your_entity'].state === 'state 2' ? 'color if state 2' : 'color if none or none'} !important;
}
.bubble-icon-container {
  background: ${hass.states['fan.your_entity'].state === 'state 1' ? 'color if state 1' : 
  hass.states['fan.your_entity'].state === 'state 2' ? 'color if state 2' : 'color if none or none'} !important;
}
```

And I'm sorry but I'm not planning to work on new themes, Bubble Card is already taking most of my free time to be honest, and a baby is coming soon! This is the main reason that pushes me to finish Bubble Card 2, my time is counted!

**Reply by willliamchan** [reply]
thanks for the code and congratulations Clooos! I dunno anything about ccs code to be honest and it took me some time to mess around with your theme to make it more into the color template that i need. is it ok if you can response once more with actual state (on/off) and any random color so I know exactly what needs to be replaced?

**Reply by Clooos** [reply]
The one I've send before can have 3 different states/colors, but if you just want to change the background and the icon when it's on you can do that:

```yaml
.bubble-button-card-container {
  background: ${hass.states['fan.your_entity'].state === 'on' ? 'rgb(40, 190, 120)' : ''} !important;
}
.bubble-icon-container {
  background: ${hass.states['fan.your_entity'].state === 'on' ? 'rgb(40, 190, 120)' : ''} !important;
}
```

**Reply by willliamchan** [reply]
.bubble-button-card-container {
  background: ${hass.states['fan.your_entity'].state === 'state 1' ? 'color if state 1' : 
  hass.states['fan.your_entity'].state === 'state 2' ? 'color if state 2' : 'color if none or none'} !important;
  hass.states['fan.your_entity'].state === 'state 3' ? 'color if state 3' : 'color if none or none'} !important;
}
.bubble-icon-container {
  background: ${hass.states['fan.your_entity'].state === 'state 1' ? 'color if state 1' : 
  hass.states['fan.your_entity'].state === 'state 2' ? 'color if state 2' : 'color if none or none'} !important;
  hass.states['fan.your_entity'].state === 'state 3' ? 'color if state 3' : 'color if none or none'} !important;
}

I see. So is that mean if I need to cover more states I can do the above?

**Reply by Clooos** [reply, marked as answer]
Yes, you can also change the color based on an attribute this way, by replacing `hass.states['fan.your_entity'].state === 'state 1'` with `hass.states['fan.your_entity'].attributes.your_attribute === 'attribute state'`. 

But I will cover all of this in the new documentation, I still have a lot of work to do on that point. I really want to make everything clear and easy to understand.

Edit: You made a mistake, it's like this:

```yaml
.bubble-button-card-container {
  background: ${hass.states['fan.your_entity'].state === 'state 1' ? 'color if state 1' :
  hass.states['fan.your_entity'].state === 'state 2' ? 'color if state 2' : 
  hass.states['fan.your_entity'].state === 'state 3' ? 'color if state 3' : 'color if none or '' '} !important;
}
```

**Reply by willliamchan** [reply]
Gotcha!

**Comment by willliamchan** [comment]
.bubble-button-card-container {
  background: ${hass.states['fan.your_entity'].state === 'on' ? 'rgb(40, 190, 120)' : ''} !important;
}
.bubble-icon-container {
  background: ${hass.states['fan.your_entity'].state === 'on' ? 'rgb(40, 190, 120)' : ''} !important;
}

Hi there! A follow up question here, how can I change the color of an icon?

**Reply by Clooos** [reply]
My mistake indeed, it's like that:
```yaml
.bubble-icon {
  color: ${hass.states['fan.your_entity'].state === 'on' ? 'rgb(40, 190, 120)' : ''} !important;
}
```

**Reply by willliamchan** [reply]
Hey thanks, this is sort of working. I was trying to put #FFFFFF (white) but no change, while when I put other color like red green using hex code it recognizes it. Any idea?

**Reply by Clooos** [reply]
Can you share your not working code?

**Reply by willliamchan** [reply]
Sure, attached two examples. The first one is to set icon color to #FFFFFF as a test to make the icon with color white. And the second example is to test to make the icon with the same color as the card container color.

![image](https://github.com/Clooos/Bubble-Card/assets/54702221/a368dbf1-32f7-406c-99f5-2e8bfc2bffd4)
.bubble-button-card-container {
  background: ${hass.states['lock.study_lock'].state === 'unlocked' ? '#6d9886' : ''} !important;
}
.bubble-icon {
  color: ${hass.states['lock.study_lock'].state === 'unlocked' ? '#FFFFFF' : ''} !important;
}
${state === 'unlocked' ? icon.setAttribute("icon", "mdi:lock-open") : icon.setAttribute("icon", "mdi:lock")};

![image](https://github.com/Clooos/Bubble-Card/assets/54702221/5b1ab4e9-2c45-46ab-83ab-59f571c1ca5b)
.bubble-button-card-container {
  background: ${hass.states['lock.study_lock'].state === 'unlocked' ? '#6d9886' : ''} !important;
}
.bubble-icon {
  color: ${hass.states['lock.study_lock'].state === 'unlocked' ? '#6d9886' : ''} !important;
}
${state === 'unlocked' ? icon.setAttribute("icon", "mdi:lock-open") : icon.setAttribute("icon", "mdi:lock")};

**Reply by Clooos** [reply]
This is because there is a default opacity on the icon, you can remove it this way:

```yaml
.bubble-icon {
  opacity: 1;
  color: ${hass.states['lock.study_lock'].state === 'unlocked' ? '#6d9886' : ''} !important;
}
```

**Reply by willliamchan** [reply]
Yea!
![Uploading image.png…]()

**Comment by willliamchan** [comment]
> ```yaml
> opacity: 1;
> ```
